### PR TITLE
[Linux/LLDB] Implement SetValue(s) functions from IDebugRegisters interface in LLDB plugin

### PR DESCRIPTION
**Pull request by kvochko**
This PR implements SetValue, SetValues, GetIndexByName functions from IDebugRegisters interface in libsosplugin.
@mikem8361 @sam-baravy @Dmitri-Botcharnikov

### CODE REVIEW

**Comment by mikem8361** [comment]
I'm curious why you are adding these functions when they are not being used by SOS? Even the ifdef'ed SOS code doesn't use the SetValue(s) functions and even where GetIndexByName is used I replaced it with one call to GetValueByName.

File: src/ToolBox/SOS/lldbplugin/services.h [hunk 1]
```diff
@@ -252,6 +253,10 @@ class LLDBServices : public ILLDBServices
     // IDebugRegisters
     //----------------------------------------------------------------------------
 
+    HRESULT GetIndexByName(
```
Comment (mikem8361) [inline src/ToolBox/SOS/lldbplugin/services.h]: ILLDBServices needs to treated like a regular COM interface. Methods can not be added like this. You need to create a ILLDBServices2 interface and put these new methods in that interface.  They can still be implemented in services.cpp but this new ILLDBServices2 needs to be added to the QueryInterface.

On the SOS side (which your changes don't seem to include any usage of this new methods) in the dbgeng.h wrapper, the ILLDBServices2 needs to be QI'ed and if it doesn't exist, they wrapper functions for the IDebugRegisters methods need to fail.

**Comment by kvochko** [comment]
@mikem8361 This is for a reverse debugging feature we are developing, for which I will make a PR soon. We thought that it might be a useful addition on its own, but if not, I could close this PR and include these changes with the feature. Also, thank you for the comments, I have separated the new methods into ILLDBServices2 interface.

**Comment by kvochko** [comment]
This feature will be implemented in a separate repository, as discussed in #7037

**Pull request closed without merging**